packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/calibrate-utils.ts

```diff
@@ -76,3 +79,49 @@ export function getErrorData(
 	});
 	return errors;
 }
+
+export const mapModelVarToDatasetVar = (state: CalibrationOperationStateCiemss, modelVariable: string) => {
+	if (modelVariable === 'timepoint_id') return state.timestampColName ?? '';
+	return state.mapping.find((d) => d.modelVariable === modelVariable)?.datasetVariable || '';
+};
+
+export function usePreparedChartInputs(
+	props: {
+		node: WorkflowNode<CalibrationOperationStateCiemss>;
+	},
+	runResult: Ref<DataArray>,
+	runResultSummary: Ref<DataArray>,
+	runResultPre: Ref<DataArray>,
+	runResultSummaryPre: Ref<DataArray>
+) {
+	const pyciemssMap = computed(() => (!runResult.value.length ? {} : parsePyCiemssMap(runResult.value[0])));
+
+	return computed(() => {
+		const state = props.node.state;
```

Comment:
I think this is fine since the output state is loaded as active state in this case and we are fetching output data with active state properties. 

We have bigger problem though. It seems once output states are saved and appended as new output, they are not updated afterward. Any state changes made on the active state don't get saved back to the state for the corresponding output. Which is a problem because once chart settings are saved after the run, you can no longer change it. When the output is on active state, you can modify it temporarily as long as it's active but when you select to different output and back to the original output, the new changes are gone and you are back to the original state that's created when you ran it.   

I think we should address the issue in a separate task.

Reply:
It was my understanding that the output state was intended to be more static as it is though I may be mistaken

Reply:
Yes, it seems fine as it is for now, but it might lead to unexpected behaviour when updating chart settings while switching between outputs. I think it should be okay for now unless someone complains... 😅  It's been like this since we introduced the output settings panel.